blockprotocol/blockprotocol
# Implement "change payment method" settings panel

## 🌟 What is the purpose of this PR?




This PR adds a "payment methods" panel to the "billing" settings tab, where users can:
- add a new payment method
- change their default subscription payment method
- remove an existing payment method
- update the billing address of an existing payment method

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203543026679373/1203781148500075/f) _(internal)_

## 🔍 What does this change?




See commits.

## ⚠️ Known issues



## 🐾 Next steps



- [Implement "change payment method" modal in the "upgrade subscription" page](https://app.asana.com/0/0/1203880493107349/f)
- [Allow users to associate a "Legal Entity" and a "Tax ID"  with an existing payment method](https://app.asana.com/0/0/1203880493107350/f)

## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Checkout the branch
2. Add the `NEXT_PUBLIC_BILLING_FEATURE_FLAG=1` env var to your `app/site/.env.local` to enable the billing feature flag
3. Run the local developer environment for the BP site
4. Checkout the HASH internal API repo **on the `bw/delete-payment-method` branch** (if it hasn't merged yet)
5. Follow the instructions in the internal API README to run the local developer environment, except for the mongo db external service (as the BP site is already running mongo db)
6. Login as a user and navigate to `/settings/billing`
7. Ensure the logged in user is subscribed to a paid subscription tier (by upgrading to "hobby" or "pro" for example)
8. Click the "Change payment method" button to make changes to the existing payment method, or add new payment method using [this list of stripe test payment details](https://stripe.com/docs/testing?testing-method=card-numbers)

## 📹 Demo




https://user-images.githubusercontent.com/42802102/216580059-f0eec2bb-9bdf-40ff-8468-314f4d627db9.mov
